docs/advanced/document-store/index.md

```diff
@@ -564,6 +564,41 @@ The solution is simple: use {ref}`column-selector<bulk-access>`:
 da[0, 'text'] = 'hello'
 ```
 
+### Performance Issue caused by List-like structure
+DocArray allows list-like behavior by adding list_like structure to storage backends. Such feature(adding list_like structure) means the database stores, 
+along with documents, meta information about document order.
+However, list_like behavior is not useful in indexers where concurrent usage is possible and users do not need information about document location. 
+Besides, updating list_like operation comes with a cost.
+You can disable list_like behavior in the config as follows
+```python
+from docarray import DocumentArray
+
+da = DocumentArray(storage='annlite', config={'n_dim': 2, 'list_like': False})
+```
+
+When list_like is disabled, all the list-like operations will not be allowed and raise errors.
+like this:
+```python
+from docarray import DocumentArray, Document
+import numpy as np
+
+
+def docs():
+    d1 = Document(embedding=np.array([10, 0]))
+    d2 = Document(embedding=np.array([0, 10]))
+    d3 = Document(embedding=np.array([-10, -10]))
+    yield d1, d2, d3
+
+
+da = DocumentArray(docs, storage='annlite', config={'n_dim': 2, 'list_like': False})
+da[0]  # This will raise an error.
+```
+**Hint**
```

Comment:
we need to follow the guidelines here: https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md#documentation-guidelines
![image](https://user-images.githubusercontent.com/15269265/200849910-5a37a652-bb25-4fc1-bc1a-ceb03e9a6893.png)